lib/python/qmk/cli/new/keyboard.py

```diff
@@ -0,0 +1,144 @@
+    valid_keyboard_name = re.compile(r'^[a-z0-9][a-z0-9_/]+$')
+
+    if cli.args.keyboard:
+        keyboard = keyboard_name = cli.args.keyboard
+    else:
+        # ask for user input if keyboard was not provided in the command line
+        cli.echo("""What is the keyboard's project name?
+
+    This will be the name used to compile firmware for your keyboard. The
+    project name may only contain lower-case letters, numbers, underscores, and
+    the forward slash, and must begin with a lower-case letter or a number.
+
+    Files will be placed in `qmk_firmware/keyboards/<project_name>/`.""")
+        keyboard = question("Project Name: ")
+
+        while valid_keyboard_name.match(keyboard) == None:
+            cli.echo(keyboard + " is not a valid project name.")
+            keyboard = question("Project Name: ")
+
+    if cli.args.project:
+        keyboard_name = cli.args.project
+    else:
+        # Ask the common name of the keyboard, if not specified by --project
+        cli.echo("""What is the keyboard's common name?
+
+    This is the name that people will use to refer to your keyboard. It should
+    be something human-friendly, like \"Clueboard 66%%\" or \"Ergodox EZ\".""")
+        keyboard_name = question("Common Name: ")
+
+
+    if cli.args.microcontroller:
+        mcu = cli.args.microcontroller
+        # Currently, AVRs are lowercase and STMs are uppercase. If the makefiles
+        #   are reworked to change the letter case used internally,
+        #   lib/python/qmk/constants.py must be updated to match.
+        if mcu.lower() in SUPPORTED_MCUS:
+            mcu = mcu.lower()
+            arch = SUPPORTED_MCUS[mcu]
+        elif mcu.upper() in SUPPORTED_MCUS:
+            mcu = mcu.upper()
+            arch = SUPPORTED_MCUS[mcu]
+        else:
+            cli.log.error(mcu + " is not a valid microcontroller option.")
+            print("  Valid Options: ", ", ".join(SUPPORTED_MCUS) )
+            exit(1)
+    else:
+        # Ask what microcontroller is being used
+        cli.echo("Select the microcontroller used:\n")
+        for i, mcu in enumerate(SUPPORTED_MCUS, 1):
+            cli.echo("    %s: %s (%s)", str(i).rjust(2, " "), mcu, SUPPORTED_MCUS[mcu] )
+
+        mcu = question("Microcontroller: (1-%d): ", len(SUPPORTED_MCUS), answer_type=int)
+        # user-facing text is 1-indexed, but data is 0-indexed internally
+        mcu -= 1
+        # create a list of tuples from the dictionary
+        mcu_list = [(name, SUPPORTED_MCUS[name]) for name in SUPPORTED_MCUS]
+        arch = mcu_list[mcu][1]
+        mcu = mcu_list[mcu][0]
+    # Set the path to the MCU architecture's template files
+    template_arch_path = template_root_path / arch
+
+
+    # (Erovia) introducing the user.name variable would probably be better
+    if cli.config.new_keyboard.username:
+        user_name = cli.config.new_keyboard.username
+    elif cli.args.username:
+        user_name = cli.args.username
```

Comment:
This is what I had in mind:

```suggestion
    if cli.config.new_keyboard.name:
        user_name = cli.config.new_keyboard.name
```

This would require renaming this subcommands argument from `--username` to `--name`.
If the user sets either `qmk config user.name=<username>` or `qmk config.new_keyboard.name` this will pick it up.
If a value is provided on the command line, it will take precedence.

Reply:
So it seems we already introduced `user.name` (https://docs.qmk.fm/#/cli_configuration?id=user-configuration-options).

So this can be safely commited.